jsprit-examples/src/main/java/com/graphhopper/jsprit/examples/DynamicServiceTimeExample.java

```diff
@@ -0,0 +1,156 @@
+            .setPickupLocation(Location.Builder.newInstance().setId("shipment:pickup").setCoordinate(Coordinate.newInstance(2, 2)).build())
+            .setDeliveryLocation(Location.Builder.newInstance().setId("shipment:dropoff").setCoordinate(Coordinate.newInstance(4, 4)).build())
+            .addPickupTimeWindow(earliestPickupTime.getEpochSecond(), latestPickupTime.getEpochSecond())
+            .addDeliveryTimeWindow(earliestDeliveryTime.getEpochSecond(), latestDeliveryTime.getEpochSecond())
+            .setDeliveryServiceTime(120D)
+            .setPickupServiceTime(120D)
+            .build();
+
+        Instant newEarliestPickupTime = Instant.parse("2020-10-05T12:00:00Z");
+        Instant newLatestPickupTime = Instant.parse("2020-10-05T12:05:00Z");
+        Instant newEarliestDeliveryTime = Instant.parse("2020-10-05T12:01:00Z");
+        Instant newLatestDeliveryTime = Instant.parse("2020-10-05T12:10:00Z");
+        Shipment newShipment = Shipment.Builder.newInstance("new")
+            .addSizeDimension(0, 1)
+            .setPickupLocation(Location.Builder.newInstance().setId("new:pickup").setCoordinate(Coordinate.newInstance(2, 2)).build())
+            .setDeliveryLocation(Location.Builder.newInstance().setId("new:dropoff").setCoordinate(Coordinate.newInstance(4, 4)).build())
+            .addPickupTimeWindow(newEarliestPickupTime.getEpochSecond(), newLatestPickupTime.getEpochSecond())
+            .addDeliveryTimeWindow(newEarliestDeliveryTime.getEpochSecond(), newLatestDeliveryTime.getEpochSecond())
+            .setDeliveryServiceTime(60D)
+            .setPickupServiceTime(60D)
+            .build();
+
+        VehicleRoute vehicleRoute = VehicleRoute.Builder.newInstance(vehicle)
+            .addPickup(shipment)
+            .addDelivery(shipment)
+            .build();
+
+        VehicleRoutingProblem vrp = VehicleRoutingProblem.Builder.newInstance()
+            .setFleetSize(VehicleRoutingProblem.FleetSize.FINITE)
+            .setRoutingCost(costMatrix)
+            .addInitialVehicleRoute(vehicleRoute)
+            .addJob(newShipment)
+            .build();
+
+        SolutionCostCalculator objectiveFunction = new SolutionCostCalculator() {
+            @Override
+            public double getCosts(VehicleRoutingProblemSolution solution) {
+                double costs = 0;
+                for (VehicleRoute route : solution.getRoutes()) {
+                    for (TourActivity activity : route.getActivities()) {
+                        costs += vrp.getActivityCosts().getActivityCost(activity, activity.getArrTime(), route.getDriver(), route.getVehicle());
+                    }
+                }
+                return costs;
+            }
+        };
+
+        StateManager stateManager = new StateManager(vrp);
+        stateManager.addStateUpdater(new UpdateActivityTimes(vrp.getTransportCosts(),ActivityTimeTracker.ActivityPolicy.AS_SOON_AS_TIME_WINDOW_OPENS_WITHIN_GROUP,  vrp.getActivityCosts()));
```

Comment:
I think this is currently obsolete if this one is used https://github.com/graphhopper/jsprit/blob/a2d67c857eb96a6faad72576160b2d0496db99d8/jsprit-core/src/main/java/com/graphhopper/jsprit/core/algorithm/PrettyAlgorithmBuilder.java#L86-L88 and I changed the AlgorithmUtil in my branch.